pymeasure/instruments/eurotest/eurotestHPP120256.py

```diff
@@ -0,0 +1,417 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+import logging
+import math
+
+import re
+import time
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_range
+from pymeasure.instruments.validators import strict_discrete_set
+
+from enum import IntFlag
+
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class EurotestHPP120256(Instrument):
+    """ Represents the Euro Test High Voltage DC Source model HPP-120-256
+    and provides a high-level interface for interacting with the instrument using the
+    Euro Test command set (Not SCPI command set).
+
+    .. code-block:: python
+
+    hpp120256 = EurotestHPP120256("GPIB0::20::INSTR")
+
+    response = hpp120256.id
+    print(response)
+    print(hpp120256.lam_status)
+    print(hpp120256.status)
+
+    hpp120256.ramp_to_zero(100.0)
+
+    hpp120256.voltage_ramp = 50.0  # V/s
+    hpp120256.current_limit = 2.0  # mA
+    inst.enable_kill = True  # Enable over-current protection
+    time.sleep(1.0)  # Give time to enable kill
+    inst.enable_output = True
+    time.sleep(1.0)  # Give time to output on
+
+    abs_output_voltage_error = 0.02 # kV
+
+    hpp120256.wait_for_output_voltage_reached(abs_output_voltage_error, 1.0, 40.0)
+
+    # Here voltage HV output should be at 0.0 kV
+
+    print("Setting the output voltage to 1.0kV...")
+    hpp120256.voltage_setpoint = 1.0  # kV
+
+    # Now HV output should be rising to reach the 1.0kV at 50.0 V/s
+
+    hpp120256.wait_for_output_voltage_reached(abs_output_voltage_error, 1.0, 40.0)
+
+    # Here voltage HV output should be at 1.0 kV
+
+    hpp120256.shutdown(100.0)
+
+    hpp120256.wait_for_output_voltage_reached(abs_output_voltage_error, 1.0, 60.0)
+
+    # Here voltage HV output should be at 0.0 kV
+
+    inst.enable_output = False
+
+    # Now the HV voltage source is in safe state
+
+    """
+
+    VOLTAGE_RANGE = [0.0, 12.0]  # kVolts
+    CURRENT_RANGE = [0.0, 25.0]  # mAmps
+    VOLTAGE_RAMP_RANGE = [10, 3000]  # V/s
+    COMMAND_DELAY = 0.2  # s
+
+    response_encoding = "iso-8859-2"
+    f_numbers_regex_pattern = r'([+-]?([\d]*\.)?[\d]+)'
+    regex = re.compile(f_numbers_regex_pattern)
```

Comment:
```suggestion
    regex = re.compile(r'([+-]?([\d]*\.)?[\d]+)')
```
If you put the string directly into the call, you avoid having a module-level variable lying around that you are actually not using elsewhere (afaict).

Reply:
Got it